packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -380,6 +406,19 @@ export class AppSyncModelVisitor<
     };
   }
 
+  InputObjectTypeDefinition(node: InputObjectTypeDefinitionNode) {
+    if (this.typesToSkip.includes(node.name.value)) {
+      return;
+    }
+    const inputValues = (node.fields as unknown) as CodeGenInputValue[];
```

Comment:
Is this type casting always known to succeed? Can the `node.fields` be undefined?

Reply:
This type casting code is similar to the line within `ObjectTypeDefinition`
https://github.com/aws-amplify/amplify-codegen/blob/a94649ef5cbed1091e4c206852d85f4b860a3eae/packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts#L310-L315
This will succeed as the type was casted by `FieldTypeDefinition`, which is before the `ObjectTypeDefinition` as the fields are the leaf node of that object. This is similar to what we have here for `InputObjectTypeDefinition`, whose fields are casted by `InputValueDefinition`